File: Prototypes/GUI Concept .NET-C++/Source Code/Example Project/Car.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Management.Instrumentation;
using System.Net.Configuration;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Example_Project
{
    struct CarStorage
    {
        public string make;
        public string model;
        public string modelNumber;
    }

    public class Car
    {
        private CarStorage cs;

        public Car(string make, string model, string modelNumber)
        {
            cs.make = make;
            cs.model = model;
            cs.modelNumber = modelNumber;
        }

        public string ModeSelecter(int currentMode)
        {
            if (currentMode == 0)
                return "Thanks for driving the " + cs.make + " " + cs.model + ".\nModel Number:" + cs.modelNumber;

            if (currentMode == 1)
                return "Welcome to your " + cs.make + " " + cs.model + ".\nModel Number:" + cs.modelNumber;

            return "Speed: 0MPH | Milage: 1200MI";
        }
    }
}

File: Prototypes/GUI Concept .NET-C++/Source Code/DLL Library/Car.cpp
#include "Car.h"
#include <string>

Car::Car(const char* make, const char* model, const char* modelNumber)
{
	cs.make = make;
	cs.model = model;
	cs.modelNumber = modelNumber;
}


std::string Car::ModeSelector(int currentMode)
{
	std::string make = cs.make, model = cs.model, modelNumber = cs.modelNumber;
	
	if (currentMode == 0)
		return std::string("Thanks for driving the " + make + " " + model + "\nModel Number:" + modelNumber);

	if (currentMode == 1)
		return std::string("Welcome to your " + make + " " + model + "\nModel Number:" + modelNumber);

	return "Speed: 0MPH | Millage: 87000MI";
}
